### PR TITLE
Ethar

### PR DESCRIPTION
**Pull request by Etharabdou10**
task6

### CODE REVIEW

**Review by Nora-Lootfy — APPROVED** [review]
Good job ya Ethar, you have nailed it

File: app/Http/Controllers/ClassController.php [hunk 1]
```diff
@@ -31,32 +33,37 @@ public function create()
...
-            'class_name'=>$request->class_name,
-            'capacity'=>$request->capacity,
-            'price'=>$request->price,
-            'timeFrom'=>$request->timeFrom,
-            'timeTo'=>$request->timeTo,
-            'is_fulled'=>isset($request->is_fulled),
+        $data=$request->validate([
+               
+            'class_name'=>'required|string',
+            'capacity'=>'required|string|max:50',
```
Comment (Nora-Lootfy) [inline app/Http/Controllers/ClassController.php:39]: capacity is not a string it is integer number `number of students class can take`

**Review by Nora-Lootfy — APPROVED** [review]
Remember ya Ethar to close pull request and merge with main `after i review your code` , so your code appear into main branch

File: resources/views/products.blade.php [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+                    <div class="row">
+                        
+                        <div class="col-12 text-center">
+                            <h2 class="mb-5">Featured Products</h2>
+                        </div>
+
+                        <div class="col-lg-4 col-12 mb-3">
+                            <div class="product-thumb">
+                                <a href="product-detail.html">
+                                    <img src="{{asset('assets/images/' .$product->image ) }}" class="img-fluid product-image" alt="">
```
Comment (Nora-Lootfy) [inline resources/views/products.blade.php:231]: i think you have mistaken here, i couldn't find your foreach

DM me if you fix it

**Review by Nora-Lootfy — CHANGES_REQUESTED** [review]
a little bug in your view file, try to fix it

File: app/Http/Controllers/ProductsController.php [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+     * Display a listing of the resource.
+     */
+    public function index()
+    {
+        $products=Products::get();
+        return view('products',compact('products'));
+        // $products = Products::latest()->take(3)->get();
+        // return view('products', compact('products'));
+    }
+    public function about()
```
Comment (Nora-Lootfy) [inline app/Http/Controllers/ProductsController.php:25]: personally, i don't like to put the public pages in the crud controller and i will tell you why

if you open the about page, you will find a section called meet our team, is this section belongs to products?

The way i prefer to work with is that and a send it as text file in the group called organizeFiles

if your website has two sections
Public pages ==> like the template of Fashion
Admin pages ==> like the crud of cars, products ...etc

for controllers
- i separate my controllers in directory called `Admin` for crud controllers or any pages for admin only
- for the public pages, it depends on the pages if there are a lot of pages i make more than one controller each one serve part of public pages & if the number of pages is small, i make one controller e.g. `PublicController` and make function for each page 
      - for example
 ```php
class PublicController {
    public function index() {
        // for index page
    }
    public function about() {
        // for about page
     }
    ...
}
```

File: app/Http/Controllers/ProductsController.php [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    {
+        $data=$request->validate([
+               
+             'title'=>'required|string',
+             'shortDescription'=>'required|string|max:1000',
+             'price'=>'required|decimal:0,1',
+             'image'=> 'required|mimes:png,jpg,jpeg|max:2048',
+             
+          ]);
+          $data['image']=$this->uploadFile($request->image,'assets/images');
```
Comment (Nora-Lootfy) [inline app/Http/Controllers/ProductsController.php:59]: you upload files to `assets/images`

File: database/factories/ProductsFactory.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+     *
+     * @return array<string, mixed>
+     */
+    public function definition(): array
+    {
+        return [
+            'title'=>fake()->randomElement(['dress', 'glasses', 't-shirt ']),
+             'shortDescription'=> fake()->text(),
+             'price'=>fake()->randomFloat(2),
+             'image'=>basename(fake()->image(public_path('assets/images/products'))),
```
Comment (Nora-Lootfy) [inline database/factories/ProductsFactory.php:23]: so why when you generate fake data upload to different directory, doesn't make sense

File: resources/views/edit_product.blade.php [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+          </div>
+          <div class="form-group mb-3 row">
+            <label for="" class="form-label col-md-2 fw-bold text-md-end">Image:</label>
+            <div class="col-md-10">
+              <input type="file" id="image" class="form-control" name="image"  >
+              @error('image')
+              <div class="alert alert-warning">{{$message}}</div>
+              @enderror
+
+                <img src="{{asset('assets/images/' .$product->image ) }}" alt="{{$product['title']}}"   style="max-width:150px"/>
```
Comment (Nora-Lootfy) [inline resources/views/edit_product.blade.php:66]: what if you are trying to edit fake generated data you will not see an image here as it is in a different directory

**Review by Nora-Lootfy — CHANGES_REQUESTED** [review]
could you please close the pull request after modifying the images path, there is a green button called `merge pull request`

you must open pull request each task, it is hard to review your code

**Review by Etharabdou10 — COMMENTED** [review]
@Nora-Lootfy

**Comment by Nora-Lootfy** [comment]
don't reopen it ya @Etharabdou10, merge the pull request then open a new pull request each time

File: app/Http/Controllers/CarController.php [hunk 7]
```diff
@@ -4,49 +4,58 @@
...
-           'price'=>$request->price,
-           'published'=>isset($request->published),
-           
+        $data=$request->validate([
+               
+          'carTitle'=>'required|string',
+           'description'=>'required|string|max:1000',
+           'price'=>'required|decimal:0,1',
+           'image'=> 'required|mimes:png,jpg,jpeg|max:2048',
+           'category_id' => 'required|exists:categories,id',
```
Comment (Nora-Lootfy) [inline app/Http/Controllers/CarController.php:48]: nice one

File: resources/views/edit_car.blade.php [hunk 8]
```diff
@@ -24,32 +24,79 @@
...
+
+          <div class="form-group mb-3 row">
+            <label for="" class="form-label col-md-2 fw-bold text-md-end">Category:</label>
+            <div class="col-md-10">
+              <select name="category_id" id="" class="form-control py-2" >
+                
+              
+
+              @foreach ($categories as $category)
+        <option value="{{ $category->id }}" @selected(old('category_id') == $category->id || (isset($car->category_id) && $car->category_id == $category->id))>{{ $category->category_name }}</option>
```
Comment (Nora-Lootfy) [inline resources/views/edit_car.blade.php:57]: try this
```php
@selected(old('category_id', $car->category_id) == $category->id)
```

File: resources/views/edit_car.blade.php [hunk 9]
```diff
@@ -24,32 +24,79 @@
...
+             @endforeach
+
+
+                </select>
+                @error('category_name')
```
Comment (Nora-Lootfy) [inline resources/views/edit_car.blade.php:62]: category_id not category_name

File: resources/views/add_car.blade.php [hunk 10]
```diff
@@ -24,33 +24,71 @@
...
+            <label for="" class="form-label col-md-2 fw-bold text-md-end">Category:</label>
             <div class="col-md-10">
-              <input type="number" step="0.1" placeholder="Enter price" class="form-control py-2" name="price"/>
+              <select name="category_id" id="" class="form-control py-2" >
+                <option value="">Select Category</option>
+               
+
+                @foreach($categories as $category )
+                
+                <option value="{{$category->id}}">{{$category->category_name}}</option>
```
Comment (Nora-Lootfy) [inline resources/views/add_car.blade.php:57]: try this
```php
@selected(old('category_id') == $category->id)
```

**Review by Nora-Lootfy — APPROVED** [review]
Merge the pull request😭

**Pull request merged**